### DISCUSSION TITLE
Utilizing hacs-hubitat to pass through Samsung HubiThings to HA

### DISCUSSION THREAD

**Discussion by sonkatech** [jason0x43/hacs-hubitat#291] [Q&A]
Hey everyone, thank you Jason for this awesome integration!

I bought a Hubitat to pass through Samsung appliances and an older TV to Home Assistant without opening it to the cloud. I'm currently able to view the devices correctly (they're on two separate VLANs) and can see the names of the sensors. I cannot see the values however.

I created a web hook manually and allowed ports and allow rules to and from the Hubitat and Home assistant (bridged already in Proxmox) but still can't get any values to appear like they do in Hubitat.

Any ideas what I can do to get entities from Hubitat to appear with the information correcty in HA? Much thanks all!

Here's my code in the configurations.yaml on the HA side:]

hubitat:
  token: !secret hubitat_token
  host: http://10.10.20.113
  port: 35629
  event_url: http://10.10.10.135:35629

![ScreenShot-10-202024 7@2x](https://github.com/user-attachments/assets/e53d805e-9bd6-4728-8598-83f5df9e33b8)
![ScreenShot-10-202024 6@2x](https://github.com/user-attachments/assets/bb536354-a581-4b24-9bdc-009c2f4037a6)
![ScreenShot-10-202024 5@2x](https://github.com/user-attachments/assets/a847318e-4df4-4eac-9914-030b62c5c353)
![ScreenShot-10-202024 4@2x](https://github.com/user-attachments/assets/a92b401a-3c01-430f-bfa5-178339ebfe0c)

**Comment by jason0x43** [comment]
Can you enable debug logging for the integration and see if any events are being sent to Home Assistant when the devices update in Hubitat?

**Comment by sonkatech** [comment]
Here's what I'm seeing for the logs when enabled for Hubitat. Do I have this setup correctly for debugging?

![ScreenShot-10-212024 3@2x](https://github.com/user-attachments/assets/029f34b4-cf6e-49c6-a194-10006b40b3e4)

![ScreenShot-10-212024 2@2x](https://github.com/user-attachments/assets/add93b9d-8953-470b-b03d-e1160251d301)

**Comment by jason0x43** [comment]
Oops, I meant to link to the instructions in the readme: [logging](https://github.com/jason0x43/hacs-hubitat?tab=readme-ov-file#logging). When it’s enabled, you’ll get a lot of Hubitat-specific messages in the log.

**Comment by sonkatech** [comment]
Gotcha, thanks for clarifying. Appreciate the quick support! Here's what I'm seeing: 

![ScreenShot-10-212024 5@2x](https://github.com/user-attachments/assets/6990413d-7d8e-4060-9f8a-3bd13c4a4456)

![ScreenShot-10-212024 4@2x](https://github.com/user-attachments/assets/b6f7fc12-8569-4395-b0fe-2d39409a2e9b)

**Comment by jason0x43** [comment]
Ok, all of that looks normal. Now try doing something on one of the devices that should trigger a state update, like opening the oven door. Check that the state updates in Hubitat, then check the HA logs and see if you see an event coming from Hubitat for whatever you did.

**Comment by sonkatech** [comment]
Great to hear. I can confirm attributes from the Hubitat side are updating on their and on the smart things app when opening the doors/ turning on the stove top. 

Here's what I'm seeing in the logs now: 

![ScreenShot-10-212024 6@2x](https://github.com/user-attachments/assets/792854c0-0c55-4c9d-a16e-5f6fd32b1005)


For what it's worth, I can turn off and on the TV using this control in HA. Also, just confirming, no need for HA to be on the same VLAN since communication is working per the logs, correct?

My end goal is not to control the appliances but to just see their states (stove top on, how much washer/ dryer time remaining, etc). I do love having local TV controls through Hubitat and hope to create a dashboard/ scripts for changing inputs/ volume/ etc in HA and HomeKit.

Thanks

![ScreenShot-10-212024 6@2x](https://github.com/user-attachments/assets/f88e152b-50cc-4e8d-9ef2-36c5a2eb075d)

**Reply by jason0x43** [reply]
Do you ever seen any event messages in the log when you use one of the devices that's having issues? For example, if you open the door on your oven, I'm guessing the `doorState` value in Hubitat changes. At the same time, you should see an event in the Home Assistant log that looks something like this (but for an oven door, not a light):

```
2024-09-15 21:29:38.715 DEBUG (MainThread) [custom_components.hubitat.hubitatmaker.hub] Received event: {'name': 'level', 'value': '30', 'displayName': 'Virtual CT Light', 'deviceId': '3', 'descriptionText': 'Virtual CT Light level was set to 30%', 'unit': '%', 'type': None, 'data': None}
2024-09-15 21:29:38.715 DEBUG (MainThread) [custom_components.hubitat.hubitatmaker.hub] Updating level of 3 to 30 (%)
```

Basically, if Hubitat is sending device events, you should be seeing `Received event:` messages in the Home Assistant from the hubitat integration.

> Also, just confirming, no need for HA to be on the same VLAN since communication is working per the logs, correct?

Yes, if _any_ devices from Hubitat are updating properly in Home Assistant (all the time, not just at startup), then everything is hooked up properly. As far as the Hubitat<->Home Assistant connection goes, there's nothing device-specific there.

**Comment by sonkatech** [comment]
Makes sense. For Received event: I'm only seeing code for the washing machine and TV (no oven even though it's actively used):


- 2024-10-22 08:57:48.616 DEBUG (MainThread) [custom_components.hubitat.hubitatmaker.hub] Received event: {'name': 'washerJobState', 'value': 'rinse', 'displayName': 'Washer - main', 'deviceId': '45', 'descriptionText': None, 'unit': None, 'type': None, 'data': None}

- 2024-10-22 08:30:38.440 DEBUG (MainThread) [custom_components.hubitat.hubitatmaker.hub] Received event: {'name': 'machineState', 'value': 'run', 'displayName': 'Washer - main', 'deviceId': '45', 'descriptionText': None, 'unit': None, 'type': None, 'data': None}

- 2024-10-22 08:30:34.870 DEBUG (MainThread) [custom_components.hubitat.hubitatmaker.hub] Received event: {'name': 'timeRemaining', 'value': '00:56:01', 'displayName': 'Washer - main', 'deviceId': '45', 'descriptionText': None, 'unit': None, 'type': None, 'data': None}

- 2024-10-22 01:49:42.866 DEBUG (MainThread) [custom_components.hubitat.hubitatmaker.hub] Received event: {'name': 'switch', 'value': 'off', 'displayName': 'Samsung TV', 'deviceId': '40', 'descriptionText': None, 'unit': None, 'type': None, 'data': None}


and for the Updating level command the following:

- 2024-10-21 19:03:23.195 DEBUG (MainThread) [custom_components.hubitat.hubitatmaker.hub] Updating switch of 40 to on (None)

- 2024-10-22 08:57:48.616 DEBUG (MainThread) [custom_components.hubitat.hubitatmaker.hub] Updating washerJobState of 45 to rinse (None)

- 2024-10-22 08:30:39.244 DEBUG (MainThread) [custom_components.hubitat.hubitatmaker.hub] Updating washerJobState of 45 to weightSensing (None)

- 2024-10-22 08:30:34.809 DEBUG (MainThread) [custom_components.hubitat.hubitatmaker.hub] Updating completionTime of 45 to 2024-10-22T13:26:34Z (None)

- 2024-10-22 08:30:34.870 DEBUG (MainThread) [custom_components.hubitat.hubitatmaker.hub] Updating timeRemaining of 45 to 00:56:01 (None)

**Comment by sonkatech** [comment]
When the cooktop is on in the stove/ range and it's confirmed on Hubitat, this is the only sensor I see appear in HA logs:

2024-10-21 19:02:13.846 DEBUG (MainThread) [custom_components.hubitat.sensor] Adding generic sensor for 44:cooktopOperatingState

![ScreenShot-10-222024@2x](https://github.com/user-attachments/assets/9c5ae0e2-1646-4542-8489-649aa8c0448d)

**Comment by sonkatech** [comment]
Is there a way to include the entities from the HubiThings Github to be understood by the hacs-hubitat integration?
https://community.hubitat.com/t/release-hubithings-replica/107976https://github.com/DaveGut/HubithingsReplica/blob/main/README.md

**Comment by jason0x43** [comment]
For the event messages, are there any accompanying messages saying that the integration updated something? Like, in this pair, there’s an event, and then a message saying the corresponding entity is being updated:
```
2024-09-15 21:29:38.715 DEBUG (MainThread) [custom_components.hubitat.hubitatmaker.hub] Received event: {'name': 'level', 'value': '30', 'displayName': 'Virtual CT Light', 'deviceId': '3', 'descriptionText': 'Virtual CT Light level was set to 30%', 'unit': '%', 'type': None, 'data': None}
2024-09-15 21:29:38.715 DEBUG (MainThread) [custom_components.hubitat.hubitatmaker.hub] Updating level of 3 to 30 (%)
```

You’re getting sensors in HA, and you’re getting events from some of the devices, but it’s not clear yet why entities aren’t being updated in HA.

Speaking of that… your original screenshots are cut off a bit.

<img width="898" alt="image" src="https://github.com/user-attachments/assets/b6fad7e0-07a8-482a-a868-028ff96c27bd">

In the shot above, is the list of sensors a list of currently active sensors, or is it a list that showed up after tapping a “+X entities not shown” link?

<img width="434" alt="image" src="https://github.com/user-attachments/assets/b74e1ed1-e617-40f3-b1be-937771ac9525">

<img width="383" alt="image" src="https://github.com/user-attachments/assets/9ccc5536-e49a-4a9f-9556-f6b3468f533b">

Regarding Hubithings GitHub, it depends on how Hubithings works. This integration depends on Hubitat’s Maker API. If Hubithings creates devices in Hubitat that behave like normal devices (e.g., they emit events through the Maker API and can receive commands through the Maker API), it _should_ work. Just out of curiosity, is there a reason you’re running the devices through Hubitat rather than connecting them directly to Home Assistant?

**Comment by sonkatech** [comment]
Thanks for the breakdown. I'm using Hubitat because the current setup with Home Assistant requires opening it to the internet and the  current drivers with Hubitat are more up to date and allow for local TV control. I found your HACS integration and thought this would be the best and cheapest way (picked up a C7 for $20 from eBay) to get local TV control while getting keeping my HA setup fully local and accessible through VPN and Apple's HomeKit.

I'm planning to spin up a new HA instance and set it up just with hacs-hubitat this weekend to see if there's something messed up with my current setup.

Here's what I've found with the Received/ Updating logs:

2024-10-23 19:52:58.655 DEBUG (MainThread) [custom_components.hubitat.hubitatmaker.hub] Received event: {'name': 'washerJobState', 'value': 'finish', 'displayName': 'Washer - main', 'deviceId': '45', 'descriptionText': None, 'unit': None, 'type': None, 'data': None}
2024-10-23 19:52:58.655 DEBUG (MainThread) [custom_components.hubitat.hubitatmaker.hub] Updating washerJobState of 45 to finish (None)

![ScreenShot-10-232024 11@2x](https://github.com/user-attachments/assets/0e2a4925-760b-4fb7-9294-8d5eec72958a)
![ScreenShot-10-232024 12@2x](https://github.com/user-attachments/assets/253c4c9b-4b5f-48e9-964f-936df9899ba9)

**Reply by jason0x43** [reply]
> I'm using Hubitat because the current setup with Home Assistant requires opening it to the internet and the current drivers with Hubitat are more up to date and allow for local TV control.

Interesting. Why would you need to open HA to the internet, but not Hubitat?

In any case, it looks like events are being handled properly. I think the issue is probably that the sensors in question simply aren't enabled. If you look at these sensors, the top two are enabled while the bottom two are not.
 
<img width="260" alt="Screenshot 2024-10-23 at 21 44 40" src="https://github.com/user-attachments/assets/e2c8084c-f27f-445d-b71d-f2d5fae71453">

In your screenshots, it looks like none of the sensor entities are enabled (which makes sense, because only disabled sensors are collapsed into the "+X entities not shown" link). If you tap one of them, you should get a popup like this:

<img width="592" alt="Screenshot 2024-10-23 at 21 50 23" src="https://github.com/user-attachments/assets/10916648-371c-4909-bdc2-ead534296624">

Tap the gear in the upper right and you'll see, among other controls, a toggle to enable the sensor.

**Comment by sonkatech** [comment]
Looks like that was the fix! No idea why I complicated this so much - thank you for your help!!

Samsung has decided to push all smart things through the cloud so it's understandable the api needs internet access but my TV is a generation 1 'smart' TV so Hubitat drivers support it well.

I will report back how this setup works as there are a lot of folks who don't want to expose their HA instances to the internet just for SmartThings API support. I think you'll have many more folks flocking to this integration once I share it on the forms!

https://www.home-assistant.io/integrations/smartthings/
https://www.home-assistant.io/integrations/samsungtv/
